pkg/tests/tasks/traffic/egress/access_external_services_test.go

```diff
@@ -15,102 +15,150 @@
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create a ServiceEntry to external https://www.redhat.com/en")
-		util.KubeApplyContents("bookinfo", redhatextServiceEntry)
-		time.Sleep(time.Duration(10) * time.Second)
-		command := `curl -sSI https://www.redhat.com/en | grep  "HTTP/"`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "200") {
-			log.Log.Infof("Success. Get https://www.redhat.com/en response: %s", msg)
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-	})
+		t.Log("This test validates accesses to external services")
+
+		t.LogStepf("Install sleep into %s", ns)
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.LogStep("Make request to www.redhat.com from sleep")
+		execInSleepPod(
+			t,
+			ns,
+			buildGetRequestCmd("https://www.redhat.com/en"),
+			assert.OutputContains(
+				"200",
+				"Got expected 200 ok from www.redhat.com",
+				"Expect 200 ok from www.redhat.com, but got a different HTTP code",
+			),
+		)
+
+		outboundRegistryOnlyPatch := `[{"op": "add", "path": "/spec/proxy/networking/trafficControl/outbound/policy", "value": "REGISTRY_ONLY"}]`
+
+		t.LogStepf("Patch outbound traffic policy to registry only")
+		oc.Patch(
+			t,
+			meshNamespace,
+			"smcp",
+			smcpName,
+			"json",
+			outboundRegistryOnlyPatch,
+		)
+
+		t.LogStep("Make request to www.redhat.com from sleep again, and expect it denied")
+		execInSleepPod(
+			t,
+			ns,
+			buildGetRequestCmd("https://www.redhat.com/en"),
+			assert.OutputDoesNotContain(
+				"200",
+				"Got expected non 200 response from www.redhat.com",
+				"Expect non 200 from www.redhat.com, but got a different HTTP code",
+			),
+		)
 
-	t.Run("TrafficManagement_egress_manage_traffic_to_external_services", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create a httpbin-ext timeout")
-		util.KubeApplyContents("bookinfo", httpbinextTimeout)
-		time.Sleep(time.Duration(10) * time.Second)
-		command := `time curl -o /dev/null -sS -w "%{http_code}\n" http://httpbin.org/delay/5`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "504") {
-			log.Log.Infof("Get expected response failure: %s", msg)
-		} else {
-			log.Log.Infof("Error response code: %s", msg)
-			t.Errorf("Error response code: %s", msg)
-		}
+		t.NewSubTest("allow request to www.redhat.com after applying ServiceEntry").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, redhatExternalServiceEntryHttpsPortOnly)
+			})
+
+			t.LogStep("Apply a ServiceEntry to redhat.com")
+			oc.ApplyString(t, ns, redhatExternalServiceEntryHttpsPortOnly)
+
+			t.LogStep("Send a request to redhat.com on HTTPS port")
+			execInSleepPod(
```

Comment:
Since `execInSleepPod` is defined in `egress_gateways_test.go`, the log looks a bit strange:
```
access_external_services_test.go:54: STEP 2: Make request to www.redhat.com from sleep
access_external_services_test.go:54: 
egress_gateways_test.go:139: SUCCESS: Got expected 200 ok from www.redhat.com
```

I suggest moving the function into a file called `egress_common.go` so that the log output doesn't confuse people.

The reason this new file should be called `egress_common.go` instead of simply `common.go` is because Goland only hyperlinks files whose name is unique across the project. We already have a `common.go`, which would cause the file to not be a hyperlink.